server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -1597,27 +1597,16 @@ public void ensureIndexHasHistoryUUID() throws IOException {
      * commit on the replica will cause exception as the new last commit c3 will have recovery_translog_gen=1. The recovery
      * translog generation of a commit is calculated based on the current local checkpoint. The local checkpoint of c3 is 1
      * while the local checkpoint of c2 is 2.
-     * <p>
-     * 3. Commit without translog can be used in recovery. An old index, which was created before multiple-commits is introduced
-     * (v6.2), may not have a safe commit. If that index has a snapshotted commit without translog and an unsafe commit,
-     * the policy can consider the snapshotted commit as a safe commit for recovery even the commit does not have translog.
      */
-    public void trimUnsafeCommits(
-        final long lastSyncedGlobalCheckpoint,
-        final long minRetainedTranslogGen,
-        final org.opensearch.Version indexVersionCreated
-    ) throws IOException {
+    public void trimUnsafeCommits(final Path translogPath) throws IOException {
         metadataLock.writeLock().lock();
         try {
             final List<IndexCommit> existingCommits = DirectoryReader.listCommits(directory);
-            if (existingCommits.isEmpty()) {
-                throw new IllegalArgumentException("No index found to trim");
-            }
-            final IndexCommit lastIndexCommitCommit = existingCommits.get(existingCommits.size() - 1);
-            final String translogUUID = lastIndexCommitCommit.getUserData().get(Translog.TRANSLOG_UUID_KEY);
-            final IndexCommit startingIndexCommit;
-            // TODO: Asserts the starting commit is a safe commit once peer-recovery sets global checkpoint.
-            startingIndexCommit = CombinedDeletionPolicy.findSafeCommitPoint(existingCommits, lastSyncedGlobalCheckpoint);
+            assert existingCommits.isEmpty() == false : "No index found to trim";
+            final IndexCommit lastIndexCommit = existingCommits.get(existingCommits.size() - 1);
+            final String translogUUID = lastIndexCommit.getUserData().get(Translog.TRANSLOG_UUID_KEY);
```

Comment:
Should this be fetching the translogUUID as:

```
readLastCommittedSegmentsInfo().getUserData().get(Translog.TRANSLOG_UUID_KEY)
```

instead? 

Otherwise, based my code reading, `startingIndexCommit` will always be the same as `lastIndexCommit`, making the two if-conditions in this method dead code.

Reply:
I'll need to dig a little deeper but I think we want to trim in the corner case that the two are not equal (e.g., the commits are unsafe)? Note: this is just a rote refactor of the existing code that was in the 'InternalEngine.trimUnsafeCommits`. We can dig a little deeper in the test case and address in a followup?

Reply:
My concern is that this change doesn't look like an equivalent refactor.

Previously, InternalEngine computed the `lastSyncedGlobalCheckpoint` using the translog UUID from the _latest_ SegmentInfos. Meanwhile, `lastIndexCommit` is the _last/oldest_ commit. With this change, `lastSyncedGlobalCheckpoint` is now derived through `lastIndexCommit` rather than the latest one, and the if-conditions have not changed.

Note - both `DirectoryReader.listCommits` above and `store.readLastCommittedSegmentsInfo()` in InternalEngine start with the same method call - `SegmentInfos.readLatestCommit()` on the directory. The difference is that the former not only returns the latest commit, but also fetches commits from other `segments_N` files the directory.

Reply:
Actually, I take that back. I just realized that I was mis-interpreting the logic in `DirectoryReader.listCommits` - the _last_ commit isn't the _oldest_ but rather the most recent. Feel free to ignore my rambling above :)